pulp_container/tests/functional/api/test_rbac_repo_versions.py

```diff
@@ -0,0 +1,269 @@
+            return repository.latest_version_href
+
+        repository = self.repositories_api.read(self.repository.pulp_href)
+        with self.assertRaises(ApiException):
+            self.user_helpless["repo_version_api"].delete(repository.latest_version_href)
+        with self.assertRaises(ApiException):
+            self.user_reader["repo_version_api"].delete(repository.latest_version_href)
+
+        response = self.repo_version_api.delete(create_new_repo_version())
+        monitor_task(response.task)
+
+        response = self.user_creator["repo_version_api"].delete(create_new_repo_version())
+        monitor_task(response.task)
+
+        response = self.user_repov_remover["repo_version_api"].delete(create_new_repo_version())
+        monitor_task(response.task)
+
+        response = self.user_repo_remover["repo_version_api"].delete(create_new_repo_version())
+        monitor_task(response.task)
+
+
+class PushRepoVersionTestCase(unittest.TestCase):
+    """Verify RBAC for repo versions of a ContainerPushRepository."""
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Define APIs to use and pull images needed later in tests
+        """
+        api_client = gen_container_client()
+        cls.pushrepository_api = RepositoriesContainerPushApi(api_client)
+        cls.namespace_api = PulpContainerNamespacesApi(api_client)
+        cls.repo_version_api = RepositoriesContainerVersionsApi(api_client)
+
+        cfg = config.get_config()
+        cls.registry = cli.RegistryClient(cfg)
+        cls.registry.raise_if_unsupported(unittest.SkipTest, "Tests require podman/docker")
+        cls.registry_name = urlparse(cfg.get_base_url()).netloc
+
+        admin_user, admin_password = cfg.pulp_auth
+        cls.user_admin = {"username": admin_user, "password": admin_password}
+        cls.user_creator = gen_user(
+            [
+                "container.add_containerdistribution",
+                "container.add_containernamespace",
+            ]
+        )
+        cls.user_reader = gen_user(["container.view_containerpushrepository"])
+        cls.user_helpless = gen_user([])
+
+        # create a push repo
+        image_path = f"{DOCKERHUB_PULP_FIXTURE_1}:manifest_d"
+        cls._pull(image_path)
+        repo_name = "testrv/perms"
+        local_url = "/".join([cls.registry_name, f"{repo_name}:1.0"])
+        cls._push(image_path, local_url, cls.user_creator)
+        cls.repository = cls.pushrepository_api.list(name=repo_name).results[0]
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete api users and things created in setUpclass."""
+        namespace = cls.namespace_api.list(name="testrv").results[0]
+        cls.namespace_api.delete(namespace.pulp_href)
+        delete_orphans()
+        del_user(cls.user_creator)
+        del_user(cls.user_reader)
+        del_user(cls.user_helpless)
+
+    @classmethod
+    def _pull(cls, image_path, user=None):
```

Comment:
but i was thinking to move this into the registry base test so we do not copy paste them every time when we write a new module with pull/push tests.